ddd-koans/Value Object/Terms Dictonary/src/Main.kt

```diff
@@ -0,0 +1,43 @@
+sealed class TimeSignature {
+    companion object {
+        private val timeSignatureRegex = "^(\\d{1,2})/(\\d{1,2})$".toRegex()
+
+        fun create(numberOfBeats: Int, noteValue: Int): TimeSignature =
+            runCatching { ValidTimeSignature(NumberOfBeats(numberOfBeats), NoteValue(noteValue)) }
+                .getOrDefault(InvalidTimeSignature)
+
+        fun of(timeSignature: String): TimeSignature =
+            timeSignatureRegex.matchEntire(timeSignature)
+                ?.groupValues
+                ?.takeIf { it.size == 3 }
+                ?.let { group -> group[1] to group[2] }
+                ?.let { (numberOfBeats, noteValue) -> create(numberOfBeats.toInt(), noteValue.toInt()) }
+                ?: InvalidTimeSignature
+    }
+}
+
```

Comment:
this shouldn't be present here I guess, it looks like a previous task about Functional approach